### PR TITLE
Remove support for cache format selection, and simplify storage related code

### PR DESCRIPTION
**Pull request by ace-e4s**
## Description
This PR aims to simplify storage and cache handling code. Key changes are:
* Deprecate `DirectDownload` and `FileCacheDownload`. Logic is moved to `BaseDownloader` and `Storage`.
    * `BaseDownloader` is to be deprecated next 👍 
* Rename `FileCacheDownload` -> `StorageCache`
* `StorageCache` deals only with local cache handling. "Try cache, then download and write to cache" logic moved to `BaseDownloader._download_verified_chunk`
* `Storage` initiates necessary cache stuff when needed.
* `Client.__init__` is significantly cleaned up as `DirectDownload` and `FileCacheDownload` is deprecated.
* Support for choosing cache file format deprecated. Uses `parquet`. Period. 

## Checklist
- [ ] PR title is descriptive and fit for injection into release notes (see tips below)
- [ ] Correct label(s) are used

PR title tips:
* Use imperative mood
* Describe the motivation for change, issue that has been solved or what has been improved - not how
* Examples:
  * Add functionality for Allan variance to sensor_4s.simulate
  * Upgrade to support Python 9.10
  * Remove MacOS from CI

### CODE REVIEW

**Comment by ace-e4s** [comment]
As before, unit-tests are janky. Hit it hard with testing!

**Comment by anna-follestad-4ss** [comment]
client.append() does not work (nothing gets appended)

File: datareservoirio/storage/storage.py [hunk 1]
```diff
@@ -26,26 +26,35 @@ class Storage:
...
-        timeseries_api: TimeseriesAPI
-            Instance of timeseries API.
-        downloader: cls
-            A strategy instance for handling downloads.
-        session : cls
-            An authenticated session that is used in all API calls. Must supply a
-            valid bearer token to all API calls.
-
-
+            Initialize service for working with timeseries data in Azure Blob
```
Comment (anna-follestad-4ss) [inline datareservoirio/storage/storage.py:31]: un-tab?

File: datareservoirio/storage/storage.py [hunk 2]
```diff
@@ -257,7 +266,7 @@ def reset_cache(self):
         """Reset the cache, deleting any stored data."""
         self._evict_entry_root(self.cache_root)
 
-    def get(self, chunk):
+    def get(self, chunk):  # "Counterpart to DirectDownload.get"
         """
```
Comment (vegard-solum-4ss) [inline datareservoirio/storage/storage.py:270]: We have deleted DirectDownload...

**Review by vegard-solum-4ss — APPROVED** [review]
Looks good!

File: datareservoirio/client.py [hunk 3]
```diff
@@ -32,41 +32,33 @@ class Client:
...
-        downloader = BaseDownloader(download_backend)
+        # TODO: Remove after 2023-08-15
+        if cache:
+            try:
+                del cache_opt["format"]
+            except (TypeError, KeyError):
+                pass
+            else:
+                warnings.warn(
+                    "Support for choosing cache format depraceted. 'format' will be ignored.",
```
Comment (ace-e4s) [inline datareservoirio/client.py:55]:
```suggestion
                    "Support for choosing cache format deprecated. 'format' will be ignored.",
```

File: datareservoirio/storage/storage.py [hunk 4]
```diff
@@ -257,7 +266,7 @@ def reset_cache(self):
         """Reset the cache, deleting any stored data."""
         self._evict_entry_root(self.cache_root)
 
-    def get(self, chunk):
+    def get(self, chunk):  # "Counterpart to DirectDownload.get"
```
Comment (ace-e4s) [inline datareservoirio/storage/storage.py:269]:
```suggestion
    def get(self, chunk):

```

**Pull request merged**